osu.Game/Skinning/Editor/SkinEditorOverlay.cs

```diff
@@ -114,15 +116,36 @@ public void OnReleased(KeyBindingReleaseEvent<GlobalAction> e)
         }
 
         /// <summary>
-        /// Exit any existing skin editor due to the game state changing.
+        /// Set a new target screen which will be used to find skinnable components.
         /// </summary>
-        public void Reset()
+        public void SetTarget(Screen screen)
         {
-            skinEditor?.Save();
-            skinEditor?.Hide();
-            skinEditor?.Expire();
+            if (skinEditor == null) return;
+
+            skinEditor.Save();
+
+            // AddOnce with parameter will ensure the newest target is loaded if there is any overlap.
+            Scheduler.AddOnce(setTarget, screen);
+        }
 
-            skinEditor = null;
+        private void setTarget(Screen target)
+        {
+            Debug.Assert(skinEditor != null);
+
+            if (!target.IsLoaded)
```

Comment:
A bit worried about this leaking delegates... Worth adding a `if (!target.IsCurrentScreen()) return;` maybe?

Reply:
Which delegate are you concerned about? The `AddOnce` call a few lines down should be fine due to the `Once` part, I think. Worst case there would only be one running.

I'll add the check anyway since it makes sense.

Reply:
As in, what happens if a screen is pushed but exited before it's ever loaded? `ScreenStack.ScreenPushed` is invoked immediately upon `Push()`.

Reply:
Yeah, there might be a loop running until the next screen change, but it'd be resolved at that point due to how `AddOnce` works.

Reply:
Ah, I see.